cmake/modules/FindHostTools.cmake

```diff
@@ -114,3 +114,6 @@ set_ifndef(TOOLCHAIN_KCONFIG_DIR ${TOOLCHAIN_ROOT}/cmake/toolchain/${ZEPHYR_TOOL
 
 set(HostTools_FOUND TRUE)
 set(HOSTTOOLS_FOUND TRUE)
+build_info(toolchain name VALUE ${ZEPHYR_TOOLCHAIN_VARIANT})
+string(TOUPPER ${ZEPHYR_TOOLCHAIN_VARIANT} zephyr_toolchain_variant_upper)
+build_info(toolchain path VALUE "${${zephyr_toolchain_variant_upper}_TOOLCHAIN_PATH}")
```

Comment:
Causing old compilation errors as ???_TOOLCHAIN_PATH is invalid. 

> -- Found toolchain: gnuarmemb (C:\gnu_arm_embedded_10)
> CMake Error at C:/Users/BillyBob/zephyrGoodX/zephyrNew/cmake/modules/yaml.cmake:313 (string):
>   string sub-command JSON failed parsing json string: * Line 1, Column 1
> 
>     Bad escape sequence in string
> 
>   See Line 1, Column 6 for detail.


I changed it to:
```
build_info(toolchain path VALUE "${TOOLCHAIN_ROOT}")
# build_info(toolchain path VALUE "${${zephyr_toolchain_variant_upper}_TOOLCHAIN_PATH}")
```

Reply:
Hey @WilliamGFish, could you open a pull request with the fix? This is the [contribution guideline documentation](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-guidelines).

Reply:
a couple of comments regarding this, first related to the issue itself.

Please note that a CMake path officially is with forward slashes `/`, and that windows path separator in principle should be escaped.
This info seems to be a bit hidden in various places of the docs, like here:
https://cmake.org/cmake/help/latest/command/cmake_path.html#modification
> If `<input>` is a native path, it is converted into a cmake-style path with forward-slashes (/).
> ...
> ...
> For example:
>  `set(native_path "c:\\a\\b/..\\c")`

or in CMake questions:
https://discourse.cmake.org/t/window-env-variable-and-cmake-path/5410
> Paths should normally be written with forward slashes in CMake code, even for Windows. Alternatively, if you want a backslash, you need to escape it, so your string would have to be either `"D:/test/fmt/release"` or `"D:\\test\\fmt\\release"`.

So the use of a single back-slash in general with CMake is **not** guaranteed to work.
That said, it seems like the `find_program()` supports the use of single `\`. The `<toolchain>_TOOLCHAIN_PATH` is generally used together with `find_program()` and therefore working.

Regarding the `build_info()` / yaml feature, which uses CMake's json support under the hood, then path should of course be properly sanitized in case they contain single `\`.

Note that `TOOLCHAIN_ROOT` != `<toolchain>_TOOLCHAIN_PATH`.
One is the toolchain itself, the other points to the location of the toolchain CMake integration code.
Those two variables points to very different things.

Reply:
Fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/79647

Reply:
You're right they are two different things just needed stop compile failures. I have addressed this by replacing the \ with / in the CMake script like this:
```
string(REGEX REPLACE "\\\\" "/" toolchain_path "${${zephyr_toolchain_variant_upper}_TOOLCHAIN_PATH}")
build_info(toolchain path VALUE "${toolchain_path}")
```
As on windows machines they will continue to use backslashes in config files the will need to be replaced.